build.sbt

```diff
@@ -79,13 +79,17 @@ lazy val core = libraryProject("core")
     libraryDependencies ++= Seq(
       cats,
       catsEffect,
+      catsParse,
       fs2Io,
       log4s,
       parboiled,
       scalaReflect(scalaVersion.value) % Provided,
       vault,
     ),
     unusedCompileDependenciesFilter -= moduleFilter("org.scala-lang", "scala-reflect"),
+    mimaBinaryIssueFilters ++= Seq(
+      ProblemFilters.exclude[MissingClassProblem]("org.http4s.HttpVersion$Parser"),
+    ),
```

Comment:
We have four options here:
1. Deprecate this, and leave dead code in Dotty that would explode.
2. Deprecate and move to a scala-2 directory. We'd need an `HttpVersionCompat` mixin.
3. Remove it and grant MiMa exceptions as necessary.
4. Surrender and call this branch 0.22, which might be necessary anyway.

On Gitter, @ChristopherDavenport and I overlapped on choice 3, so I'm pursuing that here.

Reply:
3 or 4 seems like the way to go to me.

Reply:
I was in favor of 2 or 3, and Chris liked 3 or 4.

I think the thing that may force an 0.22 is several integrations not publishing for Dotty (e.g. Play, Twirl), so we have to split up the repo, and therefore the release cycles.